tests/bootstrap.php

```diff
@@ -158,12 +272,26 @@ function loadDataset() {
          foreach($inputs as $input) {
             // Resolve FK
             foreach ($input as $k => $v) {
-               if (isForeignKeyField($k) && isset($ids[$v]) && !is_numeric($v)) {
-                  $input[$k] = $ids[$v];
+//               $foreigntype = $type; // by default same type than current type (is the case of the dropdowns)
+               $foreigntype = false ;
+               $match = array() ;
+               if( isForeignKeyField($k) && (preg_match("/(.*s)_id$/", $k, $match) || preg_match("/(.*s)_id_/", $k, $match))){
+                  $foreigntype = array_pop( $match ) ;
+                  $foreigntype = getItemTypeForTable( "glpi_$foreigntype" ) ;
+               }
+               if ( $foreigntype && isset($ids[$foreigntype][$v]) && !is_numeric($v)) {
+                  $input[$k] = $ids[$foreigntype][$v];
+               } elseif ($k == 'items_id'  &&  isset( $input['itemtype'] ) && isset($ids[$input['itemtype']][$v]) && !is_numeric($v)) {
+                  $input[$k] = $ids[$input['itemtype']][$v];
+               } elseif( $foreigntype && $foreigntype != 'UNKNOWN' && !is_numeric($v) ) {
+                  // not found in ids array, then must get it from DB
```

Comment:
I will prefer to see all data in test, to only use data from test (e.g. using TU_USER instead of 'glpi')